pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -151,19 +168,38 @@ func readConfig(config io.Reader) (VSphereConfig, error) {
 func init() {
 	vclib.RegisterMetrics()
 	cloudprovider.RegisterCloudProvider(ProviderName, func(config io.Reader) (cloudprovider.Interface, error) {
+		// If vSphere.conf file is not present then it is worker node.
+		if config == nil {
+			return newWorkerNode()
+		}
 		cfg, err := readConfig(config)
 		if err != nil {
 			return nil, err
 		}
-		return newVSphere(cfg)
+		return newControlllerNode(cfg)
 	})
 }
 
 // Initialize passes a Kubernetes clientBuilder interface to the cloud provider
 func (vs *VSphere) Initialize(clientBuilder controller.ControllerClientBuilder) {}
 
-func newVSphere(cfg VSphereConfig) (*VSphere, error) {
+// Creates new worker node interface and returns
+func newWorkerNode() (*VSphere, error) {
 	var err error
+	vs := VSphere{}
+	vs.hostName, err = os.Hostname()
+	if err != nil {
+		glog.Errorf("Failed to get hostname. err: %+v", err)
+		return nil, err
+	}
+
+	return &vs, nil
+}
+
+// Creates new Contreoller node interface and returns
+func newControlllerNode(cfg VSphereConfig) (*VSphere, error) {
+	var err error
+	glog.Errorf("cfg is\n%+v\n", cfg)
```

Comment:
I have just kept it for now. We can't log cfg anyways as it will print VC credentials in logs which we don't want.
I have added TODO comment to remove it once everything is settled.